kyber/dash.py

```diff
@@ -1,52 +1,86 @@
-import click
+import json
 import os
-from urlparse import urlparse, urlunparse
-
-from kyber.context import Context, ContextError
-from kyber.utils import get_executable_path
+import subprocess
+from urlparse import urljoin
 
+import click
 
-def _kube_dash_base_url(cfg):
-    url = urlparse(cfg.cluster['server'])
-    try:
-        username = cfg.user['username']
-        password = cfg.user['password']
-        url = url._replace(netloc='{}:{}@{}'.format(username, password, url.netloc))
-    except KeyError:  # no username/pass in kube context?
-        pass
-    return url
+from kyber.context import Context, ContextError
 
 
 def service_dashboard(cfg, name):
-    path = ("/api/v1/proxy/namespaces/kube-system/services/kubernetes-dashboard/"
-            "#/service/{namespace}/{app_name}?namespace={namespace}").format(
-                namespace=cfg.namespace, app_name=name)
-    return urlunparse(_kube_dash_base_url(cfg)._replace(path=path))
+    path = (
+        "/api/v1/namespaces/kube-system/services/https:kubernetes-dashboard:"
+        "/proxy/#!/service/{namespace}/{app_name}?namespace={namespace}"
+    ).format(namespace=cfg.namespace, app_name=name)
+    return urljoin("http://localhost:8001", path)
 
 
 def namespace_dashboard(cfg):
-    path = ("/api/v1/proxy/namespaces/kube-system/services/kubernetes-dashboard/"
-            "#/pod?namespace={namespace}").format(namespace=cfg.namespace)
-    return urlunparse(_kube_dash_base_url(cfg)._replace(path=path))
+    path = (
+        "/api/v1/namespaces/kube-system/services/https:kubernetes-dashboard:"
+        "/proxy/#!/overview?namespace={namespace}"
+    ).format(namespace=cfg.namespace)
+    return urljoin("http://localhost:8001", path)
+
+
+def _get_token(cfg):
+    user = cfg.user
+    if "exec" not in user:
+        return
+
+    args = user["exec"]["args"]
+    args.insert(0, user["exec"]["command"])
+
+    try:
+        response = subprocess.check_output(args)
+        parsed = json.loads(response)
+    except Exception as e:
+        click.echo("Unable to get token ({})".format(e))
+        return
+
+    return parsed.get("status", {}).get("token")
 
 
-def launch(cfg=None, executor=None):
+def launch(cfg=None, executor=None):  # noqa: C901
     if cfg is None:
         from kyber.lib.kube import kube_api
+
         cfg = kube_api.config
+
     if executor is None:
         executor = os.execve
 
+    token = _get_token(cfg)
+    if token:
+        click.echo("Your login token:\n\n{}\n".format(token))
+
+    click.echo("Starting kubectl proxy")
+    try:
+        proxy = subprocess.Popen(["kubectl", "proxy"])
```

Comment:
I ended up opening the proxy with `subprocess.Popen`, so that I could start it before trying to open the url, then I just end up waiting for the process with `.wait()`. Doing ctrl+c at the end kills the proxy and the script finishes running.